### PR TITLE
add authz.SetProviders

### PR DESCRIPTION
**Pull request by kwangil-ha**

### CODE REVIEW

**Comment by unknwon** [comment]
Hi @kwangil-ha, could you explain why this change is needed?

**Comment by kwangil-ha** [comment]
> Hi @kwangil-ha, could you explain why this change is needed?
This is the same reason as below issue.
https://github.com/sourcegraph/sourcegraph/issues/3847

**Comment by unknwon** [comment]
Thanks, I'll trigger a CI build for this PR.

File: cmd/frontend/main.go [hunk 1]
```diff
@@ -9,5 +10,6 @@ import (
 // function for details.
 
 func main() {
+	go authz.SetProviders(true, []authz.Provider{})
```
Comment (unknwon) [inline cmd/frontend/main.go]: I think it is safe without making a goroutine (as OSS does not have authz providers nor does we watch for config changes to them):
```suggestion
	// Set dummy authz provider to unblock channel for checking permissions in GraphQL APIs.
	// See https://github.com/sourcegraph/sourcegraph/issues/3847 for details.
	authz.SetProviders(true, []authz.Provider{})
```

Comment (unknwon) [reply cmd/frontend/main.go]: cc @beyang to suggest wording here.

Comment (unknwon) [inline cmd/frontend/main.go:13]:
```suggestion
	// Set dummy authz provider to unblock channel for checking permissions in GraphQL APIs.
	// See https://github.com/sourcegraph/sourcegraph/issues/3847 for details.
	authz.SetProviders(true, []authz.Provider{})
```

**Comment by unknwon** [comment]
Closed in favor of https://github.com/sourcegraph/sourcegraph/pull/10258.

**Pull request closed without merging**